content/en/docs/concepts/storage/persistent-volumes.md

```diff
@@ -785,6 +785,65 @@ spec:
       storage: 10Gi
 ```
 
+## Volume Populators and Data Sources
+
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+{{< note >}}
+Volume populars are available as alpha since Kubernetes 1.18, are redesigned in Kubernetes 1.22. The `AnyVolumeDataSource` feature must be enabled. Refer to the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) documentation for more information.
+{{< /note >}}
+
+Volume populators take advantage of a new PVC spec field called `dataSourceRef`. Unlike the
+`dataSource` field, which can only contain either a reference to another PersistentVolumeClaim
+or to a VolumeSnapshot, the `dataSourceRef` field can contain a reference to any object in the
+same namespace, except for core objects other than PVCs.
```

Comment:
Do core objects only refer to APIs defined in https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go?
Does this mean APIs defined in other groups under https://github.com/kubernetes/kubernetes/blob/master/pkg/apis such as StatefulSet in "apps" maybe supported?

Reply:
Just core. We won't reject one with a apiGroup of "apps" -- those will simply fail to bind because nobody will write a populator for statefulsets.